### PR TITLE
Performance improvements for pagination query

### PR DESCRIPTION
**Pull request by mxaddict**
Performance improvements for pagination query

This basically fixes the performance issue when doing pagination on VERY Large data sets with a Group By.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Please send to 5.1.

**Comment by GrahamCampbell** [comment]
Please send to 5.1.

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -1643,43 +1646,38 @@ public function buildRawPaginator($paginator, $results, $perPage)
...
 		// Because some database engines may throw errors if we leave the ordering
 		// statements on the query, we will "back them up" and remove them from
 		// the query. Once we have the count we will put them back onto this.
+		$this->backupFieldsForCount($grouped);
+
+		// Add the magic FROM entry when we have grouping that needs to be
+		// done by the query
+		if ($grouped)
+		{
+			$this->from = \DB::raw("({$selectSql}) as subquery");
```
Comment (crynobone) [inline src/Illuminate/Database/Query/Builder.php]: Facade in core?

Comment (mxaddict) [inline src/Illuminate/Database/Query/Builder.php]: Yeah, I don't know of the underlying method of doing it, do you have a suggestion for a lower level replacement? :smiley:

**Comment by mxaddict** [comment]
@GrahamCampbell, I have a project that is running laravel 4.2, and needs this fix, for now I just created a fork and started using that I'n my app, I will check if 5.x has the same issue and also create a patch for it if need, but for now, would it be possible to have this approved for 4.2.x once I polish it? :smiling_imp:

**Comment by mxaddict** [comment]
@GrahamCampbell  And by polish I just mean, fix any code that does not look worthy of being the in core (Like the lines that @crynobone pointed out) and updating the testsuite.

Please let me know If you would consider a merge once I have done those things :smile_cat:

File: src/Illuminate/Database/Query/Builder.php (hunk 1 above)
Comment (franzliedke) [inline src/Illuminate/Database/Query/Builder.php]: `$this->connection->raw()` should do the trick.

Comment (mxaddict) [inline src/Illuminate/Database/Query/Builder.php]: Thanks! :+1:

**Comment by GrahamCampbell** [comment]
We're not accepting fixes to 4.2, sorry. This must go into 5.1.

**Comment by mxaddict** [comment]
@GrahamCampbell, May I know why? :disappointed:

**Comment by GrahamCampbell** [comment]
Because we don't support it anymore, and that's it.

**Comment by mxaddict** [comment]
Created new PR for 5.1 fix, Pull Request #9055

**Pull request closed without merging**